File: stratego-backend/src/main/java/com/pdrosoft/matchmaking/stratego/dto/StrategoMovementDTO.java
package com.pdrosoft.matchmaking.stratego.dto;

import com.pdrosoft.matchmaking.stratego.enums.Rank;

import jakarta.validation.constraints.Max;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class StrategoMovementDTO {

	@NotNull
	private Rank rank;

	@NotNull
	@Min(1)
	@Max(4)
	private int rowInitial;
	@NotNull
	@Min(1)
	@Max(10)
	private int colInitial;

	@NotNull
	@Min(1)
	@Max(4)
	private int rowFinal;
	@NotNull
	@Min(1)
	@Max(10)
	private int colFinal;
}

File: stratego-frontend/Assets/Scripts/comms/dto/StrategoMovementDTO.cs
using System;
using System.Collections.Generic;

[Serializable]
public class StrategoMovementDTO
{
	public Rank rank;

	public int rowInitial;
	public int colInitial;

	public int rowFinal;
	public int colFinal;

	public List<StrategoMovementResultDTO> result;

	public virtual bool Equals(StrategoMovementDTO movement)
	{
		if (movement == null)
		{
			return false;
		}

		return rank == movement.rank //
			&& rowInitial == movement.rowInitial //
			&& colInitial == movement.colInitial //
			&& rowFinal == movement.rowFinal //
			&& colFinal == movement.colFinal //
		;
	}
}
